### PR TITLE
create utils.jl; begin testing

### PR DESCRIPTION
**Pull request by AKS1996**

### CODE REVIEW

**Comment by AKS1996** [comment]
Hey @blegat @joaquimg

Any suggestions/reviews? This is my 1st time including test cases (1st time a creating package indeed.)

File: src/utils.jl [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    # set constraints
+    for i in 1:m
+        MOI.add_constraint(optimizer,MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.(A[i,:], x), 0.),MOI.LessThan(b[i]))
+    end
+
+    for i in 1:n
+        MOI.add_constraint(optimizer,MOI.SingleVariable(x[i]),MOI.GreaterThan(0.))
+    end
+end
```
Comment (matbesancon) [inline src/utils.jl:36]:
```suggestion
end

```
adding a new line at the end of file

File: test/runtests.jl [hunk 2]
```diff
@@ -1,6 +1,10 @@
...
 
-@testset "DiffOpt.jl" begin
-    # Write your own tests here.
-end
+const MOI = MathOptInterface;
+const MOIU = MathOptInterface.Utilities;
+
+@testset "utils" begin
+    include("utils.jl")
+end
```
Comment (matbesancon) [inline test/runtests.jl:10]:
```suggestion
end

```

File: test/utils.jl [hunk 3]
```diff
@@ -0,0 +1,10 @@
+using GLPK
+
+@testset "Creating an LP" begin
+    model = GLPK.Optimizer()
+    x̂ = generate_lp(model,10,5)
+    
+    MOI.optimize!(model)
+
+    @test MOI.get(model, MOI.TerminationStatus()) in [MOI.LOCALLY_SOLVED, MOI.OPTIMAL]
+end
```
Comment (matbesancon) [inline test/utils.jl:10]:
```suggestion
end

```

File: src/utils.jl [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+Note: Mutates the `optimizer` object
+"""
+function generate_lp(optimizer,n,m)
+    s = rand(m)
+    s = 2*s.-1
+    λ = max.(-s, 0)
+    s = max.(s, 0)
+    x̂ = rand(n)
+    A = rand(m, n)
+    b = A*x̂ .+ s
```
Comment (matbesancon) [inline src/utils.jl:18]:
```suggestion
    b = A * x̂ .+ s
```

File: src/utils.jl [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    c = -A'*λ;
```
Comment (matbesancon) [inline src/utils.jl:19]:
```suggestion
    c = -A' * λ
```
spacing

File: src/utils.jl [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    x = MOI.add_variables(optimizer, n)
+
+    # define objective
+    objective_function = MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.(c, x), 0.0)
+    MOI.set(optimizer, MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(), objective_function)
+    MOI.set(optimizer, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+
+    # set constraints
+    for i in 1:m
+        MOI.add_constraint(optimizer,MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.(A[i,:], x), 0.),MOI.LessThan(b[i]))
```
Comment (matbesancon) [inline src/utils.jl:30]:
```suggestion
        MOI.add_constraint(
            optimizer,
            MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.(A[i,:], x), 0.), MOI.LessThan(b[i]),
        )
```

Comment (matbesancon) [reply src/utils.jl:30]: this is mostly formatting

File: src/utils.jl [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+minimize c'x
+subject to Ax <= b, x >= 0 
+where x in R^{n}, A in R^{m*n}, b in R^{m}, c in R^{n}
+
+Note: Mutates the `optimizer` object
+"""
+function generate_lp(optimizer,n,m)
+    s = rand(m)
+    s = 2*s.-1
```
Comment (matbesancon) [inline src/utils.jl:13]:
```suggestion
    s = 2*s .- 1
```

File: src/utils.jl [hunk 8]
```diff
@@ -0,0 +1,36 @@
+"""
+Generates a non-trivial random MOI linear program by adding variables
+and constraints to MOI compatible Optimizer `optimizer`
+
+minimize c'x
```
Comment (matbesancon) [inline src/utils.jl:5]:
```suggestion
minimize `c' * x`
```

File: src/utils.jl [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+subject to Ax <= b, x >= 0
```
Comment (matbesancon) [inline src/utils.jl:6]:
```suggestion
subject to `Ax <= b, x >= 0`
```

File: src/utils.jl [hunk 10]
```diff
@@ -0,0 +1,36 @@
+"""
+Generates a non-trivial random MOI linear program by adding variables
+and constraints to MOI compatible Optimizer `optimizer`
+
+minimize c'x
+subject to Ax <= b, x >= 0 
+where x in R^{n}, A in R^{m*n}, b in R^{m}, c in R^{n}
```
Comment (matbesancon) [inline src/utils.jl:7]:
```suggestion
where `x in R^{n}, A in R^{m*n}, b in R^{m}, c in R^{n}`
```

**Comment by matbesancon** [comment]
I added few minor suggestions, overall this is a nice first step for the structure of the project

**Comment by AKS1996** [comment]
thanks!
next thing i'm gonna do is search some good Julia lint extension :smile:

**Comment by matbesancon** [comment]
One thing you can look at for JuMP is:
https://www.juliaopt.org/JuMP.jl/dev/style/
For general purpose: https://github.com/domluna/JuliaFormatter.jl

**Pull request merged**